src/internal/file.h

```diff
@@ -0,0 +1,8 @@
+#ifndef RLANG_INTERNAL_FILE_H
+#define RLANG_INTERNAL_FILE_H
+
+#include <stdio.h>
+
+FILE* r_fopen(r_obj* path, const char* mode);
```

Comment:
It would be helpful to mention that the caller needs to manage the implicit protection stack (not needed here but may be in future usages).

Reply:
Decided to manage the stack for the user with `vmaxget()` and `vmaxset()`